Feat/project study filters phs

ZakirG opened the pull request:
### New Features
- Guppy support for portal's new config option `adminPreAppliedFilters`. The filter is applied to to the records displayed before any user-applied filters, and can be specified independently in the dataExplorerConfig and fileExplorerConfig blocks. 

### Breaking Changes


### Bug Fixes


### Improvements


### Dependency updates


### Deployment changes
- Commons admins can specify a selectedValues filter using the `adminPreAppliedFilters` field in a portal config file using Guppy syntax. The filter is applied to to the records displayed before any user-applied filters, and can be specified independently in the dataExplorerConfig and fileExplorerConfig blocks. Example: 
```
"adminAppliedPreFilters": {
      "project_id": { 
        "selectedValues": ["jnkns-jenkins"]
      }
    }
```

qingyashu commented on src/components/ConnectedFilter/index.jsx:
```diff
@@ -21,25 +23,32 @@ class ConnectedFilter extends React.Component {
+      filter: Object.assign({}, this.props.adminAppliedPreFilters),
     };
     this.filterGroupRef = React.createRef();
+    this.adminPreFiltersFrozen = JSON.stringify(this.props.adminAppliedPreFilters).slice();
```
🤔  could I know why we want this frozen variable?

mfshao replied:
sometimes `props` disappears so this is kind of a work around to save and recreate it?

ZakirG replied:
yeah! some piece of code was modifying my variable, but after extensive debugging, could not locate it.

qingyashu commented on src/components/Utils/filters.js:
```diff
@@ -0,0 +1,25 @@
+/* eslint import/prefer-default-export: 0 */
```
Awesome that we separate this into a util function. Would be better to have it unit tested.

ZakirG replied:
done ! added unit tests

qingyashu commented:
![image](https://user-images.githubusercontent.com/38708936/65722943-fe5e5780-e072-11e9-89d8-5adec76561d6.png)


I tested in storybook, adding a prefilter like this: 
```
const preFilter = {
      "study": { 
        "selectedValues": [
          "DEV-test-Study-2",
          "DEV-test-Study-4",
        ]
      }
    };
```
 The initial rendering seems good but if I check one of the study I think the result should filter out 6 subjects.

ZakirG commented:
> ![image](https://user-images.githubusercontent.com/38708936/65722943-fe5e5780-e072-11e9-89d8-5adec76561d6.png)
> 
> I tested in storybook, adding a prefilter like this:
> 
> ```
> const preFilter = {
>       "study": { 
>         "selectedValues": [
>           "DEV-test-Study-2",
>           "DEV-test-Study-4",
>         ]
>       }
>     };
> ```
> 
> The initial rendering seems good but if I check one of the study I think the result should filter out 6 subjects.

Good catch!! Fixed this and deployed to qa-dcp. This time, the dataExplorerConfig contains 
```"adminAppliedPreFilters": {
        "project_id": { "selectedValues": ["jnkns-jenkins", "jnkns-jenkins2"]}
    }```
to match your storybook case, and the fileExplorerConfig contains 
```"adminAppliedPreFilters": {
        "project_id": { "selectedValues": ["jnkns-jenkins"]}
    }```
to match the original use case. I also added unit tests to test the case you caught, Qingya

mfshao commented:
update the version number in `package.json` and commit the lock?

qingyashu commented on src/components/Utils/filters.js:
```diff
@@ -0,0 +1,33 @@
+  const filterAB = Object.assign({}, userFilter);
+  Object.keys(adminAppliedPreFilter).forEach((key) => {
+    if (Object.prototype.hasOwnProperty.call(userFilter, key)
+          && Object.prototype.hasOwnProperty.call(adminAppliedPreFilter, key)) {
```
I guess this is an unnecessary check

ZakirG replied:
oh true! haha

qingyashu replied:
:p hehe just saw it merged, nvm~
